src/shr_transport.h4

```diff
@@ -0,0 +1,592 @@
+dnl vi: set ft=m4
+/* -*- C -*-
+ *
+ * Copyright (c) 2019 Intel Corporation. All rights reserved.
+ * This software is available to you under the BSD license.
+ *
+ * This file is part of the Sandia OpenSHMEM software package. For license
+ * information, see the LICENSE file in the top level directory of the
+ * distribution.
+ *
+ */
+
+/*
+ * This is a generated file, do not edit directly.
+ */
+
+include(shmem_bind_c.m4)dnl
+#
+#ifndef SHR_TRANSPORT_H
+#define SHR_TRANSPORT_H
+
+#ifdef USE_XPMEM
+#include "transport_xpmem.h"
+#endif
+
+#ifdef USE_CMA
+#include "transport_cma.h"
+#endif
+
+
+static inline int
+shmem_shr_transport_init(void)
+{
+    int ret = 0;
+
+#if USE_XPMEM
+    ret = shmem_transport_xpmem_init();
+    if (0 != ret)
+        RETURN_ERROR_MSG("XPMEM init failed (%d)\n", ret);
+
+#elif USE_CMA
+    ret = shmem_transport_cma_init();
+    if (0 != ret)
+        RETURN_ERROR_MSG("CMA init failed (%d)\n", ret);
+#endif
+
+    return ret;
+}
+
+
+static inline int
+shmem_shr_transport_startup(void)
+{
+    int ret = 0;
+
+#if USE_XPMEM
+    ret = shmem_transport_xpmem_startup();
+    if (0 != ret) {
+        RETURN_ERROR_MSG("XPMEM startup failed (%d)\n", ret);
+    }
+
+#elif USE_CMA
+    ret = shmem_transport_cma_startup();
+    if (0 != ret) {
+        RETURN_ERROR_MSG("CMA startup failed (%d)\n", ret);
+    }
+#endif
+
+    return ret;
+}
+
+
+static inline void
+shmem_shr_transport_fini(void)
+{
+#if USE_XPMEM
+    shmem_transport_xpmem_fini();
+#elif USE_CMA
+    shmem_transport_cma_fini();
+#endif
+}
+
+
+/* Return in local_ptr, a local pointer that can be used to access the target
+ * buffer for the PE with noderank ID. */
+static inline void
+shmem_shr_transport_ptr(void *target, int noderank, void **local_ptr)
+{
+#if USE_XPMEM
+    XPMEM_GET_REMOTE_ACCESS(target, noderank, *local_ptr);
+#else
+    RAISE_ERROR_MSG("No path to peer (%d)\n", noderank);
```

Comment:
Should we return NULL here instead of raising an error?

Reply:
No, this function is only used internally within the shared memory transport, after checking that the given PE is reachable via the shared transport. Unreachability in this function is therefore an error.